packages/ra-ui-materialui/src/field/types.ts

```diff
@@ -5,6 +5,10 @@ import { TableCellProps } from '@mui/material/TableCell';
 
 type TextAlign = TableCellProps['align'];
 type SortOrder = 'ASC' | 'DESC';
+export interface SortPayload {
+    field: string;
+    order: SortOrder;
+}
```

Comment:
I would recommend avoiding duplication of the `SortPayload` interface, and using the one from `ra-core` instead. It does not force the SortOrder to be `'ASC' | 'DESC'`, but actually we shouldn't force it since it's dependent on the dataProvider.